### PR TITLE
Externalize dependency-check

### PR DESCRIPTION
**Pull request by JaniruTEC**
This PR moves the dependency-check to the [skymatic/workflows](https://github.com/skymatic/workflows/) project, namely [run-dependency-check.yml](https://github.com/skymatic/workflows/blob/main/.github/workflows/run-dependency-check.yml)

Also see: https://github.com/cryptomator/cryptolib/pull/45

Current workflow SHA at time of PR: https://github.com/skymatic/workflows/commit/c83f5bb886e46951587fd9dbfb7ce6cd409d73a6

## Summary by CodeRabbit

- **Refactor**
  - Enhanced dependency check workflow with external configuration for improved maintainability and security.

### CODE REVIEW

**Comment by JaniruTEC** [comment]
@coderabbitai review

**Review by infeo — APPROVED** [review]

File: .github/workflows/dependency-check.yml [hunk 1]
```diff
@@ -10,54 +10,11 @@ on:
...
-          SLACK_CHANNEL: 'cryptomator-desktop'
-          SLACK_TITLE: "Vulnerabilities in ${{ github.event.repository.name }} detected."
-          SLACK_MESSAGE: "Download the <https://github.com/${{ github.repository }}/actions/run/${{ github.run_id }}|report> for more details."
-          SLACK_FOOTER: false
-          MSG_MINIMAL: true
-      - name: Failing workflow on release branch
-        if: github.event_name == 'push' && steps.dependency-check.outcome == 'failure'
-        shell: bash
-        run: exit 1
+    uses: skymatic/workflows/.github/workflows/run-dependency-check.yml@main
```
[the remark this answers is not in the document]
Comment (overheadhunter) [reply .github/workflows/dependency-check.yml:13]: We discussed this offline and decided to reference version-branches such as `@v1` that ensure a stable API but can receive minor version updates without the need to change all the workflow files again.
This is also the best practice used with GitHub Actions.

[the remark this answers is not in the document]
Comment (JaniruTEC) [reply .github/workflows/dependency-check.yml]: @coderabbitai Does the latest commit fix the issue?

[the remark this answers is not in the document]
Comment (JaniruTEC) [reply .github/workflows/dependency-check.yml]: @overheadhunter 
As already noted offline: The decision to use `v1` over `main` slipped through. I added an appropriate branch: https://github.com/skymatic/workflows/tree/v1

**Pull request merged**